File: src/main/java/com/example/test/domain/models/Task.java
package com.example.test.domain.models;

import com.example.test.domain.TaskStatusEnum;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.With;

import java.time.LocalDateTime;

@AllArgsConstructor
@Getter
@Builder
public class Task {
    private Integer id;
    @With
    private final String name;
    @With
    private final String description;
    @With
    private final String task;
    @With
    private TaskStatusEnum state;
    @With
    private final Customer customer;
    @With
    private final User creationUser;
    @With
    private final User modificationUser;
    @With
    private final LocalDateTime creationTime;
    @With
    private final LocalDateTime modificationTime;
}

File: src/main/resources/front/src/app/model/task.ts

export class Task {
    id: string;
    description: string = '';
    state: string ="";
    customer: {
        id: number;
        person: {
            id: number;
            name: string;
            surname: string;
            dateOfBirth: string;
        };
        creationTime: string;
    };

    constructor() {
        this.id = "";
        this.customer = {
            id: 0,
            person: {
                id: 0,
                name: "",
                surname: "",
                dateOfBirth: ""
            },
            creationTime: ""
        };
    }
}
